test/e2e/sync/sync.go

```diff
@@ -0,0 +1,449 @@
+// Copyright 2021 IBM Corp
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sync
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev2/controllerv2"
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+	"github.com/ppc64le-cloud/pvsadm/test/e2e/framework"
+	"gopkg.in/yaml.v2"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	APIKey      = string(os.Getenv("IBMCLOUD_API_KEY"))
+	FolderName  = "folder" + strings.ToLower(generateRandomString(6))
+	PlanSlice   = []string{"smart", "standard", "vault", "cold"}
+	RegionSlice = []string{"us-east", "jp-tok", "us-south"}
+)
+
+const (
+	ServiceType            = "cloud-object-storage"
+	ResourceGroupAPIRegion = "global"
+	ServicePlan            = "standard"
+	ResourceGrp            = "powervs-ipi-resource-group"
+	Debug                  = false
+	Recursive              = false
+	InstanceType           = "service_instance"
+	NoOfSources            = 2
+	NoOfTargetsPerSource   = 2
+	NoOfObjects            = 5
+)
+
+type Spec []struct {
+	Source `yaml:"source"`
+	Target `yaml:"target"`
+}
+
+type Source struct {
+	Bucket string `yaml:"bucket"`
+	Cos    string `yaml:"cos"`
+	Object string `yaml:"object"`
+	Plan   string `yaml:"plan"`
+	Region string `yaml:"region"`
+}
+
+type Target []struct {
+	Bucket string `yaml:"bucket"`
+	Plan   string `yaml:"plan"`
```

Comment:
Not sure what the UI or IBM Cloud API uses `StorageType` vs `Plan`, use whichever is relevant.

Reply:
IBM cloud has StorageClass, code changed accordingly.